File: PokerServer/src/config/CCfgKeyMgrTemplate.java
package config;

import java.util.HashMap;
import java.util.Map;

import utils.TabFile;

public class CCfgKeyMgrTemplate {
	public Map<Integer, ITabItemWithKey> m_ItemTable = new HashMap<Integer, ITabItemWithKey>();
	protected ITabItemWithKey NewItem(){
		return null;
	}
	public boolean Init(String strFileName){

        if(null != m_ItemTable && m_ItemTable.size() > 0){
        	return false;
        }
        m_ItemTable.clear();
        TabFile tf = new TabFile(strFileName, Config.ReadFileAsString(strFileName));
        while(tf.Next()){
        	ITabItemWithKey item  = NewItem();
        	if(item.ReadItem(tf) == false){
        		continue;
        	}
        	if(m_ItemTable.containsKey(item.GetKey())){
        		continue;
        	}
        	m_ItemTable.put(item.GetKey(), item);
        }
        return true;
	
		
	}
	public ITabItemWithKey GetConfig(int key){
		if(m_ItemTable.containsKey(key)){
			return m_ItemTable.get(key);
		}
		return null;
	}
	public boolean Has(int key){
		return m_ItemTable.containsKey(key);
	}
}

File: TexasPoker/Assets/Scripts/Configs/CCfgKeyMgrTemplate.cs
using Client.Base;
using Util;
using System.Collections.Generic;
using UnityEngine;
public interface ITabItemWithKey<TKey>
{
    bool ReadItem(TabFile tf);
    TKey GetKey();
}
public abstract class CCfgKeyMgrTemplate<TManager, TKey, TItem> : Singleton<TManager>
    where TManager : class, new()
    where TItem : ITabItemWithKey<TKey>, new()
{
    protected SortedList<TKey, TItem> m_ItemTable = new SortedList<TKey, TItem>();

    public virtual SortedList<TKey, TItem> ItemTable { get { return m_ItemTable; } }

    public virtual bool Init(TextAsset text)
    {
        if (null == text)
        {
            return false;
        }

        if (null != m_ItemTable && m_ItemTable.Count > 0)
        {
            return false;
        }

        m_ItemTable.Clear();

        TabFile tf = new TabFile(text.name, text.text);
        while (tf.Next())
        {
            TItem item = new TItem();
            if (item.ReadItem(tf) == false)
            {
                continue;
            }
            if (m_ItemTable.ContainsKey(item.GetKey()))
            {
                continue;
            }
            m_ItemTable.Add(item.GetKey(), item);
        }
        return true;
    }

    public virtual TItem GetConfig(TKey key)
    {
        if (m_ItemTable.ContainsKey(key))
        {
            return m_ItemTable[key];
        }
        return default(TItem);
    }

    public virtual bool Has(TKey key)
    {
        return m_ItemTable.ContainsKey(key);
    }

    public virtual TItem this[TKey key]
    {
        get { return GetConfig(key); }
    }
}